fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java

```diff
@@ -161,15 +171,17 @@ private String generate(Node node) {
         }
 
         if (hasVocabularyConstraint(elementDefinition)) {
-            return generateVocabularyConstraint(elementDefinition);
+            String expr = generateVocabularyConstraint(elementDefinition);
+            if (node.children.stream().noneMatch(child -> hasConstraint(child))) {
```

Comment:
maybe a comment to explain this line?  like why do we only add the vocabulary constraint if the child has no constraints.

Reply:
Added comments to clarify the behavior.